iceoryx_posh/include/iceoryx_posh/experimental/internal/popo/typed_publisher.inl

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_EXPERIMENTAL_POSH_POPO_TYPED_PUBLISHER_INL
+#define IOX_EXPERIMENTAL_POSH_POPO_TYPED_PUBLISHER_INL
+
+#include <cstdint>
+
+namespace iox
+{
+namespace popo
+{
+template <typename T, typename base_publisher_t>
+TypedPublisher<T, base_publisher_t>::TypedPublisher(const capro::ServiceDescription& service)
+    : base_publisher_t(service)
+{
+}
+
+template <typename T, typename base_publisher_t>
+inline uid_t TypedPublisher<T, base_publisher_t>::getUid() const noexcept
+{
+    return base_publisher_t::getUid();
+}
+
+template <typename T, typename base_publisher_t>
+inline cxx::expected<Sample<T>, AllocationError> TypedPublisher<T, base_publisher_t>::loan() noexcept
+{
+    return base_publisher_t::loan(sizeof(T));
+}
+
+template <typename T, typename base_publisher_t>
+inline void TypedPublisher<T, base_publisher_t>::publish(Sample<T> sample) noexcept
+{
+    return base_publisher_t::publish(sample);
```

Comment:
If sample is a rvalue you have to write here: `base_publisher_t::publish(std::forward<Sample<T>>(sample));`